src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -132,13 +132,21 @@ export class CodeWhispererCodeCoverageTracker {
             codewhispererUserGroup: CodeWhispererUserGroupSettings.getUserGroup().toString(),
             codewhispererCustomizationArn: selectedCustomization.arn === '' ? undefined : selectedCustomization.arn,
         })
+
+        let codewhispererRuntimeLanguage: string = this._language
+        if (this._language === 'jsx') {
+            codewhispererRuntimeLanguage = 'javascript'
+        } else if (this._language === 'tsx') {
+            codewhispererRuntimeLanguage = 'typescript'
+        }
+
```

Comment:
I think instead of changing here we may want to change the one that's something called "VSCToCodeWhispererRuntimeLanguage"

Reply:
this one is for patch only. I am refactoring the whole language mapping stuff

Reply:
Ok, when available please create such ticket and briefly describe what's going to be refactored.

Reply:
yea I will, I think I will follow similarly what we have in JB implementation